### ISSUE TITLE
ingress-nginx pod and jobs issues when deploying using rke up

### ISSUE THREAD

**Issue by dewongway** [rancher/rke#3063]
Hi, 

I'm trying to deploy a simple rke kubernetes using rke utility and while the overall deployment seems to be successful, the ingress-nginx pods and jobs are having issues. Looking into the describe output, I can see that it's complaining about the serviceaccount and secret "ingress-nginx-admission" are missing.  Does anyone know why they are not being created?  I tried both 1.3.15 and 1.4.0-rc3 and I hit the same issue in both versions.

Thanks
Henry

**RKE version:**
1.3.15

**Docker version: (`docker version`,`docker info` preferred)**
20.10.17-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
\# cat /etc/os-release 
NAME="SLES"
VERSION="15-SP4"
VERSION_ID="15.4"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP4"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp4"
DOCUMENTATION_URL="https://documentation.suse.com/"

\# uname -r
5.14.21-150400.24.21-default

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VMware VM

**cluster.yml file:**
```
# cat cluster.yml
# If you intended to deploy Kubernetes in an air-gapped environment,
# please consult the documentation on how to configure custom RKE images.
nodes:
- address: rke-4
  port: "22"
  internal_address: ""
  role:
  - controlplane
  - worker
  - etcd
  hostname_override: ""
  user: root
  docker_socket: /var/run/docker.sock
  ssh_key: ""
  ssh_key_path: ~/.ssh/id_rsa
  ssh_cert: ""
  ssh_cert_path: ""
  labels: {}
  taints: []
services:
  etcd:
    image: ""
    extra_args: {}
    extra_args_array: {}
    extra_binds: []
    extra_env: []
    win_extra_args: {}
    win_extra_args_array: {}
    win_extra_binds: []
    win_extra_env: []
    external_urls: []
    ca_cert: ""
    cert: ""
    key: ""
    path: ""
    uid: 0
    gid: 0
    snapshot: null
    retention: ""
    creation: ""
    backup_config: null
  kube-api:
    image: ""
    extra_args: {}
    extra_args_array: {}
    extra_binds: []
    extra_env: []
    win_extra_args: {}
    win_extra_args_array: {}
    win_extra_binds: []
    win_extra_env: []
    service_cluster_ip_range: 10.43.0.0/16
    service_node_port_range: ""
    pod_security_policy: false
    always_pull_images: false
    secrets_encryption_config: null
    audit_log: null
    admission_configuration: null
    event_rate_limit: null
  kube-controller:
    image: ""
    extra_args: {}
    extra_args_array: {}
    extra_binds: []
    extra_env: []
    win_extra_args: {}
    win_extra_args_array: {}
    win_extra_binds: []
    win_extra_env: []
    cluster_cidr: 10.42.0.0/16
    service_cluster_ip_range: 10.43.0.0/16
  scheduler:
    image: ""
    extra_args: {}
    extra_args_array: {}
    extra_binds: []
    extra_env: []
    win_extra_args: {}
    win_extra_args_array: {}
    win_extra_binds: []
    win_extra_env: []
  kubelet:
    image: ""
    extra_args: {}
    extra_args_array: {}
    extra_binds: []
    extra_env: []
    win_extra_args: {}
    win_extra_args_array: {}
    win_extra_binds: []
    win_extra_env: []
    cluster_domain: cluster.local
    infra_container_image: ""
    cluster_dns_server: 10.43.0.10
    fail_swap_on: false
    generate_serving_certificate: false
  kubeproxy:
    image: ""
    extra_args: {}
    extra_args_array: {}
    extra_binds: []
    extra_env: []
    win_extra_args: {}
    win_extra_args_array: {}
    win_extra_binds: []
    win_extra_env: []
network:
  plugin: canal
  options: {}
  mtu: 0
  node_selector: {}
  update_strategy: null
  tolerations: []
authentication:
  strategy: x509
  sans: []
  webhook: null
addons: ""
addons_include: []
system_images:
  etcd: rancher/mirrored-coreos-etcd:v3.5.4
  alpine: rancher/rke-tools:v0.1.87
  nginx_proxy: rancher/rke-tools:v0.1.87
  cert_downloader: rancher/rke-tools:v0.1.87
  kubernetes_services_sidecar: rancher/rke-tools:v0.1.87
  kubedns: rancher/mirrored-k8s-dns-kube-dns:1.21.1
  dnsmasq: rancher/mirrored-k8s-dns-dnsmasq-nanny:1.21.1
  kubedns_sidecar: rancher/mirrored-k8s-dns-sidecar:1.21.1
  kubedns_autoscaler: rancher/mirrored-cluster-proportional-autoscaler:1.8.5
  coredns: rancher/mirrored-coredns-coredns:1.9.3
  coredns_autoscaler: rancher/mirrored-cluster-proportional-autoscaler:1.8.5
  nodelocal: rancher/mirrored-k8s-dns-node-cache:1.21.1
  kubernetes: rancher/hyperkube:v1.24.4-rancher1
  flannel: rancher/mirrored-coreos-flannel:v0.15.1
  flannel_cni: rancher/flannel-cni:v0.3.0-rancher6
  calico_node: rancher/mirrored-calico-node:v3.22.0
  calico_cni: rancher/calico-cni:v3.22.0-rancher1
  calico_controllers: rancher/mirrored-calico-kube-controllers:v3.22.0
  calico_ctl: rancher/mirrored-calico-ctl:v3.22.0
  calico_flexvol: rancher/mirrored-calico-pod2daemon-flexvol:v3.22.0
  canal_node: rancher/mirrored-calico-node:v3.22.0
  canal_cni: rancher/calico-cni:v3.22.0-rancher1
  canal_controllers: rancher/mirrored-calico-kube-controllers:v3.22.0
  canal_flannel: rancher/mirrored-flannelcni-flannel:v0.17.0
  canal_flexvol: rancher/mirrored-calico-pod2daemon-flexvol:v3.22.0
  weave_node: weaveworks/weave-kube:2.8.1
  weave_cni: weaveworks/weave-npc:2.8.1
  pod_infra_container: rancher/mirrored-pause:3.6
  ingress: rancher/nginx-ingress-controller:nginx-1.2.1-rancher1
  ingress_backend: rancher/mirrored-nginx-ingress-controller-defaultbackend:1.5-rancher1
  ingress_webhook: rancher/mirrored-ingress-nginx-kube-webhook-certgen:v1.1.1
  metrics_server: rancher/mirrored-metrics-server:v0.6.1
  windows_pod_infra_container: rancher/mirrored-pause:3.6
  aci_cni_deploy_container: noiro/cnideploy:5.2.3.2.1d150da
  aci_host_container: noiro/aci-containers-host:5.2.3.2.1d150da
  aci_opflex_container: noiro/opflex:5.2.3.2.1d150da
  aci_mcast_container: noiro/opflex:5.2.3.2.1d150da
  aci_ovs_container: noiro/openvswitch:5.2.3.2.1d150da
  aci_controller_container: noiro/aci-containers-controller:5.2.3.2.1d150da
  aci_gbp_server_container: noiro/gbp-server:5.2.3.2.1d150da
  aci_opflex_server_container: noiro/opflex-server:5.2.3.2.1d150da
ssh_key_path: ~/.ssh/id_rsa
ssh_cert_path: ""
ssh_agent_auth: false
authorization:
  mode: none
  options: {}
ignore_docker_version: null
enable_cri_dockerd: null
kubernetes_version: ""
private_registries: []
ingress:
  default_backend: true
  default_ingress_class: true
  http_port: 0
  https_port: 0
  network_mode: none
  node_selector: {}
  provider: nginx
cluster_name: ""
cloud_provider:
  name: ""
prefix_path: ""
win_prefix_path: ""
addon_job_timeout: 0
bastion_host:
  address: ""
  port: ""
  user: ""
  ssh_key: ""
  ssh_key_path: ""
  ssh_cert: ""
  ssh_cert_path: ""
  ignore_proxy_env_vars: false
monitoring:
  provider: metrics-server
  options: {}
  node_selector: {}
  update_strategy:
    strategy: RollingUpdate
    rollingUPdate:
      maxUnavailable: 8
  replicas: null
  tolerations: []
  metrics_server_priority_class_name: ""
restore:
  restore: false
  snapshot_name: ""
rotate_encryption_key: false
dns:
  provider: coredns
  # Available as of v1.1.0
  update_strategy:
    strategy: RollingUpdate
    rollingUpdate:
      maxUnavailable: 20%
      maxSurge: 15%
  linear_autoscaler_params:
    cores_per_replica: 0.34
    nodes_per_replica: 4
    prevent_single_point_failure: true
    min: 2
    max: 3
```

**Steps to Reproduce:**
\# rke up
The command completed successfully. However, the nginx-ingress-controller pod  and ingress-nginx jobs are having issues.  It seems that it failed to create the service account ingress-nginx-admission and the secret "ingress-nginx-admission" not found

```
# kubectl get all -n ingress-nginx
NAME                                        READY   STATUS              RESTARTS   AGE
pod/default-http-backend-5c4c75685d-mwvpz   1/1     Running             0          3m31s
pod/nginx-ingress-controller-458g2          0/1     ContainerCreating   0          3m32s

NAME                                         TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
service/default-http-backend                 ClusterIP   10.43.180.18   <none>        80/TCP    3m31s
service/ingress-nginx-controller-admission   ClusterIP   10.43.187.4    <none>        443/TCP   3m32s

NAME                                      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/nginx-ingress-controller   1         1         0       1            0           <none>          3m32s

NAME                                   READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/default-http-backend   1/1     1            1           3m31s

NAME                                              DESIRED   CURRENT   READY   AGE
replicaset.apps/default-http-backend-5c4c75685d   1         1         1       3m31s

NAME                                       COMPLETIONS   DURATION   AGE
job.batch/ingress-nginx-admission-create   0/1           3m31s      3m31s
job.batch/ingress-nginx-admission-patch    0/1           3m31s      3m31s
```
```
# kubectl describe job.batch/ingress-nginx-admission-create -n ingress-nginx
Name:             ingress-nginx-admission-create
Namespace:        ingress-nginx
Selector:         controller-uid=0cd4d4e5-6d69-4bb7-8e48-a91a0502ffe9
Labels:           app.kubernetes.io/component=admission-webhook
                  app.kubernetes.io/instance=ingress-nginx
                  app.kubernetes.io/name=ingress-nginx
                  app.kubernetes.io/version=1.2.0
Annotations:      helm.sh/hook: pre-install,pre-upgrade
                  helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
Parallelism:      1
Completions:      1
Completion Mode:  NonIndexed
Start Time:       Wed, 12 Oct 2022 16:48:44 -0500
Pods Statuses:    0 Active (0 Ready) / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app.kubernetes.io/component=admission-webhook
                    app.kubernetes.io/instance=ingress-nginx
                    app.kubernetes.io/name=ingress-nginx
                    app.kubernetes.io/version=1.2.0
                    controller-uid=0cd4d4e5-6d69-4bb7-8e48-a91a0502ffe9
                    job-name=ingress-nginx-admission-create
  Service Account:  ingress-nginx-admission
  Containers:
   create:
    Image:      rancher/mirrored-ingress-nginx-kube-webhook-certgen:v1.1.1
    Port:       <none>
    Host Port:  <none>
    Args:
      create
      --host=ingress-nginx-controller-admission,ingress-nginx-controller-admission.$(POD_NAMESPACE).svc
      --namespace=$(POD_NAMESPACE)
      --secret-name=ingress-nginx-admission
    Environment:
      POD_NAMESPACE:   (v1:metadata.namespace)
    Mounts:           <none>
  Volumes:            <none>
Events:
  Type     Reason        Age                   From            Message
  ----     ------        ----                  ----            -------
  Warning  FailedCreate  2m1s (x6 over 6m51s)  job-controller  Error creating: pods "ingress-nginx-admission-create-" is forbidden: error looking up service account ingress-nginx/ingress-nginx-admission: serviceaccount "ingress-nginx-admission" not found
```
```
# kubectl describe job.batch/ingress-nginx-admission-patch -n ingress-nginx
Name:             ingress-nginx-admission-patch
Namespace:        ingress-nginx
Selector:         controller-uid=229827ed-92a4-4295-b0c8-becf389bc94f
Labels:           app.kubernetes.io/component=admission-webhook
                  app.kubernetes.io/instance=ingress-nginx
                  app.kubernetes.io/name=ingress-nginx
                  app.kubernetes.io/version=1.2.0
Annotations:      helm.sh/hook: post-install,post-upgrade
                  helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
Parallelism:      1
Completions:      1
Completion Mode:  NonIndexed
Start Time:       Wed, 12 Oct 2022 16:48:44 -0500
Pods Statuses:    0 Active (0 Ready) / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app.kubernetes.io/component=admission-webhook
                    app.kubernetes.io/instance=ingress-nginx
                    app.kubernetes.io/name=ingress-nginx
                    app.kubernetes.io/version=1.2.0
                    controller-uid=229827ed-92a4-4295-b0c8-becf389bc94f
                    job-name=ingress-nginx-admission-patch
  Service Account:  ingress-nginx-admission
  Containers:
   patch:
    Image:      rancher/mirrored-ingress-nginx-kube-webhook-certgen:v1.1.1
    Port:       <none>
    Host Port:  <none>
    Args:
      patch
      --webhook-name=ingress-nginx-admission
      --namespace=$(POD_NAMESPACE)
      --patch-mutating=false
      --secret-name=ingress-nginx-admission
      --patch-failure-policy=Fail
    Environment:
      POD_NAMESPACE:   (v1:metadata.namespace)
    Mounts:           <none>
  Volumes:            <none>
Events:
  Type     Reason        Age                    From            Message
  ----     ------        ----                   ----            -------
  Warning  FailedCreate  4m52s (x6 over 9m42s)  job-controller  Error creating: pods "ingress-nginx-admission-patch-" is forbidden: error looking up service account ingress-nginx/ingress-nginx-admission: serviceaccount "ingress-nginx-admission" not found

```

```
kubectl describe pod/nginx-ingress-controller-458g2 -n ingress-nginx
Name:           nginx-ingress-controller-458g2
Namespace:      ingress-nginx
Priority:       0
Node:           rke-4/10.10.140.83
Start Time:     Wed, 12 Oct 2022 16:48:44 -0500
Labels:         app=ingress-nginx
                app.kubernetes.io/component=controller
                app.kubernetes.io/instance=ingress-nginx
                app.kubernetes.io/name=ingress-nginx
                controller-revision-hash=5bd46759d
                pod-template-generation=1
Annotations:    <none>
Status:         Pending
IP:             
IPs:            <none>
Controlled By:  DaemonSet/nginx-ingress-controller
Containers:
  controller:
    Container ID:  
    Image:         rancher/nginx-ingress-controller:nginx-1.2.1-rancher1
    Image ID:      
    Ports:         80/TCP, 443/TCP, 8443/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP
    Args:
      /nginx-ingress-controller
      --default-backend-service=$(POD_NAMESPACE)/default-http-backend
      --election-id=ingress-controller-leader-nginx
      --controller-class=k8s.io/ingress-nginx
      --configmap=$(POD_NAMESPACE)/ingress-nginx-controller
      --tcp-services-configmap=$(POD_NAMESPACE)/tcp-services
      --udp-services-configmap=$(POD_NAMESPACE)/udp-services
      --validating-webhook=:8443
      --validating-webhook-certificate=/usr/local/certificates/cert
      --validating-webhook-key=/usr/local/certificates/key
      --watch-ingress-without-class=true
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Liveness:       http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=5
    Readiness:      http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=3
    Environment:
      POD_NAME:       nginx-ingress-controller-458g2 (v1:metadata.name)
      POD_NAMESPACE:  ingress-nginx (v1:metadata.namespace)
      LD_PRELOAD:     /usr/local/lib/libmimalloc.so
    Mounts:
      /usr/local/certificates/ from webhook-cert (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-sx289 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  webhook-cert:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ingress-nginx-admission
    Optional:    false
  kube-api-access-sx289:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 :NoExecute op=Exists
                             :NoSchedule op=Exists
                             node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists
                             node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                             node.kubernetes.io/unreachable:NoExecute op=Exists
                             node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason       Age                  From               Message
  ----     ------       ----                 ----               -------
  Normal   Scheduled    13m                  default-scheduler  Successfully assigned ingress-nginx/nginx-ingress-controller-458g2 to rke-4
  Warning  FailedMount  2m50s (x3 over 11m)  kubelet            Unable to attach or mount volumes: unmounted volumes=[webhook-cert], unattached volumes=[kube-api-access-sx289 webhook-cert]: timed out waiting for the condition
  Warning  FailedMount  98s (x14 over 13m)   kubelet            MountVolume.SetUp failed for volume "webhook-cert" : secret "ingress-nginx-admission" not found
  Warning  FailedMount  32s (x3 over 9m38s)  kubelet            Unable to attach or mount volumes: unmounted volumes=[webhook-cert], unattached volumes=[webhook-cert kube-api-access-sx289]: timed out waiting for the condition

```
**Results:**

**Comment by Josh-Diamond** [comment]
Reproduced
---

With RKE `v1.3.15` + `SLES-15-SP4 - docker-20.10.14-ce`

1. Spin up single-instance: `SLES-15-SP4 - docker-20.10.14-ce` 
2. Ssh into instance and add user to docker
   - usermod -aG docker $USER
   - systemctl start docker
   - systemctl enable docker
   - reboot
3. Using `Cluster.yml` provided in issue description, run `rke up`
4. Reproduced - ingress-nginx pod + jobs fail to create

Logs:
```
# kubectl get all -n ingress-nginx
NAME                                        READY   STATUS              RESTARTS   AGE
pod/default-http-backend-<REDACTED>         1/1     Running             0          2m29s
pod/nginx-ingress-controller-<REDACTED>     0/1     ContainerCreating   0          2m29s

NAME                                         TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
service/default-http-backend                 ClusterIP   <REDACTED>      <none>        80/TCP    2m29s
service/ingress-nginx-controller-admission   ClusterIP   <REDACTED>      <none>        443/TCP   2m29s

NAME                                      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/nginx-ingress-controller   1         1         0       1            0           <none>          2m30s

NAME                                   READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/default-http-backend   1/1     1            1           2m30s

NAME                                              DESIRED   CURRENT   READY   AGE
replicaset.apps/default-http-backend-<REDACTED>   1         1         1       2m30s

NAME                                       COMPLETIONS   DURATION   AGE
job.batch/ingress-nginx-admission-create   0/1           2m30s      2m30s
job.batch/ingress-nginx-admission-patch    0/1           2m30s      2m30s


# kubectl describe job.batch/ingress-nginx-admission-create -n ingress-nginx
Warning  FailedCreate  38s (x5 over 2m48s)  job-controller  Error creating: pods "ingress-nginx-admission-create-" is forbidden: error looking up service account ingress-nginx/ingress-nginx-admission: serviceaccount "ingress-nginx-admission" not found
```

---

Workaround
---

With RKE `v1.3.15` + `SLES-15-SP4 - docker-20.10.14-ce`

1. Spin up single-instance: `SLES-15-SP4 - docker-20.10.14-ce` 
2. Ssh into instance and add user to docker
   - usermod -aG docker $USER
   - systemctl start docker
   - systemctl enable docker
   - reboot
3. Using `Cluster.yml` provided below, run `rke up`
```yaml
ssh_key_path: <path/to/ssh/key>

nodes:
  - address: <REDACTED>
    internal_address: <REDACTED>
    user: $USER
    role: [etcd, controlplane, worker]
```

4. Verified - ingress-nginx pod + jobs successfully created

Logs:
```
# kubectl get all -n ingress-nginx
NAME                                            READY   STATUS      RESTARTS   AGE
pod/ingress-nginx-admission-create-<REDACTED>   0/1     Completed   0          51s
pod/ingress-nginx-admission-patch-<REDACTED>    0/1     Completed   0          51s
pod/nginx-ingress-controller-<REDACTED>         1/1     Running     0          52s

NAME                                         TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
service/ingress-nginx-controller-admission   ClusterIP   <REDACTED>     <none>        443/TCP   52s

NAME                                      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/nginx-ingress-controller   1         1         1       1            1           <none>          52s

NAME                                       COMPLETIONS   DURATION   AGE
job.batch/ingress-nginx-admission-create   1/1           6s         52s
job.batch/ingress-nginx-admission-patch    1/1           6s         52s
```

**Comment by dewongway** [comment]
Hi Josh,

Thanks for the solution. I have used the `cluster.yml` in your workaround in my env and it resolved the issue.

**Comment by Josh-Diamond** [comment]
@dewongway Have you been able to successfully spin up clusters, with the `cluster.yml` provided in the issues description, in the past?  I'm trying to  determine if this is a bug, or related to user error w/ mis-configurations in yaml, perhaps. 🤔

**Comment by dewongway** [comment]
@Josh-Diamond I'm new to RKE and kubernetes so it's very likely I did something wrong in the config file. I have never successfully spun up the ingress-nginx controllers and jobs using the `cluster.yml` provided in the issue description. After successfully bringing up the cluster using your `cluster.yml`, I went back to my original `cluster.yml` file and set the `internal_address` just like yours and the cluster came up.  My host `rke-4` is in DNS and I have no issue resolving to IP. That's why I use the system name in the configure instead of IP.
